robolectric/src/main/java/org/robolectric/RobolectricTestRunner.java

```diff
@@ -98,24 +100,24 @@ private void assureTestLifecycle(SdkEnvironment sdkEnvironment) {
 
   protected DependencyResolver getJarResolver() {
     if (dependencyResolver == null) {
-      if (Boolean.getBoolean("robolectric.offline")) {
-        String dependencyDir = System.getProperty("robolectric.dependency.dir", ".");
-        dependencyResolver = new LocalDependencyResolver(new File(dependencyDir));
-      } else {
-        File cacheDir = new File(new File(System.getProperty("java.io.tmpdir")), "robolectric");
-        cacheDir.mkdir();
-
-        if (cacheDir.exists()) {
-          dependencyResolver = new CachedDependencyResolver(new MavenDependencyResolver(), cacheDir, 60 * 60 * 24 * 1000);
-        } else {
-          dependencyResolver = new MavenDependencyResolver();
-        }
-      }
+      File cacheDir = new File(new File(System.getProperty("java.io.tmpdir")), "robolectric");
+      cacheDir.mkdir();
+      String dependencyDir = getLocalDependencyDir();
+      dependencyResolver = new CompositeDependencyResolver(Arrays.asList(
+              new LocalDependencyResolver(dependencyDir == null?null:new File(dependencyDir)),
+              new CachedDependencyResolver(new MavenDependencyResolver(), cacheDir, CACHE_EXPIRE_TIME)));
     }
-
     return dependencyResolver;
   }
 
+  private String getLocalDependencyDir(){
+    String dependencyDir = System.getenv("ROBOLECTRIC_DEPENDENCY_DIR");
```

Comment:
We don't currently expose any configuration via environment variables, so I'd prefer to do this in a separate commit (and potentially do it for all of robolectric's configuration).